libraries/AP_RangeFinder/AP_RangeFinder_PulsedLightLRF.cpp

```diff
@@ -90,7 +91,8 @@ void AP_RangeFinder_PulsedLightLRF::timer(void)
         if (_dev->write_register(LL40LS_MEASURE_REG, LL40LS_MSRREG_ACQUIRE)) {
             phase = PHASE_COLLECT;
         }
-        break;
+        if (!v3hp_hardware)
+            break;
```

Comment:
This doesn't seem correct to me, between asking for a new measurement and reading the distance registers, we have to wait.

I understood your point about getting faster updates and it makes sense, but it should be the other way around: after reading the distance, we can immediately ask for a new measurement.

Reply:
The issue is with that immediately there, we have threads, there is no guarantee it will happen immediately - the issue here is that you might read the two distance registers while they are being written and so the info could be incorrect.

It might work in practice 99.99% of the time, but doing it the other way around is guaranteed to not be reading registers that might be being written (not completely true since we don't check the status bit, but, given we set a minimum update rate higher than we read, it's theoretically true).